docs/tutorials/qsimcirq_gcp.md

```diff
@@ -0,0 +1,379 @@
+
+
+
+![alt_text](../images/image13.png )
+
+
+
+
+*   Choose the [Machine Type](https://cloud.google.com/blog/products/compute/choose-the-right-google-compute-engine-machine-type-for-you): n2-standard-16
+    *   16 CPUs
+    *   64GB memory
+*   Choose the Boot Disk image:[ Container-Optimized OS](https://cloud.google.com/container-optimized-os/docs/concepts/features-and-benefits)
+    *   Leave the remaining as defaults.
+
+
+
+
+![alt_text](../images/image10.png )
+
+
+
+
+Finally, enable HTTP access
+*   Click "Create"
+
+
+![alt_text](../images/image8.png )
+
+## Running the qsim docker container
+
+There are two ways to run qsim from this container: from a Jupyter Notebook and interactively from a shell. In both cases, you need to connect to the VM you created above.
+
+
+### Open a SSH window to your VM
+
+Once VM is created you can connect via [SSH](https://cloud.google.com/compute/docs/ssh-in-browser) to your newly created simulation VM. The easiest is to click the SSH button in the console from the **Compute Engine** -> **VM Instances** page:
+
+
+
+![alt_text](../images/image5.png )
+
+
+You will have a popup window connecting you to the simulation VM. You should have a command prompt:
+
+
+
+
+
+![alt_text](../images/image6.png )
+
+
+Start the container:
+
+
+```console
+    $ docker run -v `pwd`:/homedir -p 8888:8888 gcr.io/quantum-builds/github.com/quantumlib/jupyter_qsim:latest
+```
+
+
+The output will be something like:
+
+
+```console
+ Unable to find image 'gcr.io/quantum-291919/jupyter_qsim:latest' locally
+ latest: Pulling from quantum-291919/jupyter_qsim
+ 3c72a8ed6814: Pull complete 
+ 2bd71a698eae: Pull complete 
+ Digest: sha256:d0d0040c6ef9925719459e736631c9fec957b94d377009c591d6285143ebb626
+ ...    
+     To access the notebook, open this file in a browser:
+         file:///root/.local/share/jupyter/runtime/nbserver-1-open.html
+     Or copy and paste one of these URLs:
+         http://79804d33f250:8888/?token=d7a53e728e3dff08c6ed12a15810471486b4c15828a3d70a
+      or http://127.0.0.1:8888/?token=d7a53e728e3dff08c6ed12a15810471486b4c15828a3d70a
+```
+
+
+Copy the last URL (it starts with `127.0.0.1`). Replace `127.0.0.1` with the IP address of the VM 
+
+
+[created above](#build-a-container-optimized-vm): 
+
+
+```console
+ http://[EXTERNAL_IP_ADDRESS]:8888/?token=7191178ae9aa4ebe1698b07bb67dea1d289cfd0e0b960373
+```
+
+
+Paste that in your browser. You should now see the Jupyter UI:
+
+
+
+
+
+![alt_text](../images/image2.png )
+
+
+Navigate to  qsim > docs > tutorials. You will see:
+
+
+
+
+
+![alt_text](../images/image12.png )
+
+
+Click on qsimcirq.ipynb. This will load the notebook.
+
+You can skip the setup and go straight to the cell labeled "Full state-vector simulation":
+
+
+
+![alt_text](../images/image1.png )
```

Comment:
```suggestion
Copy the last URL (it starts with `127.0.0.1`). Replace `127.0.0.1` with the IP address of the VM 

[created above](#build-a-container-optimized-vm): 

```console
 http://[EXTERNAL_IP_ADDRESS]:8888/?token=7191178ae9aa4ebe1698b07bb67dea1d289cfd0e0b960373
```

Paste that in your browser. You should now see the Jupyter UI:

![alt_text](../images/image2.png )

Navigate to  qsim > docs > tutorials. You will see:

![alt_text](../images/image12.png )

Click on qsimcirq.ipynb. This will load the notebook.

You can skip the setup and go straight to the cell labeled "Full state-vector simulation":

![alt_text](../images/image1.png )
```